### PR TITLE
feat(init): set mode status

### PR DESCRIPTION
**Pull request by tapakornl**

### CODE REVIEW

File: packages/modules/init-capital/src/actions/setModeStatus.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        modeStatuses: [
+          { mode: 1, status: { canCollateralize: true, canDecollateralize: false, canBorrow: true, canRepay: false } },
+          { mode: 2, status: { canCollateralize: false, canDecollateralize: false, canBorrow: false, canRepay: false } },
+        ],
+      },
+    }
+    // data.
+    const SetModeStatusAction = new SetModeStatusActionTest(data)
+    const subActions: SetModeStatusSubAction[] = SetModeStatusAction.getSubActions() as SetModeStatusSubAction[]
+    console.log()
```
Comment (mewpawee) [inline packages/modules/init-capital/src/actions/setModeStatus.test.ts:38]: remove this

File: packages/modules/init-capital/src/actions/setModeStatus.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { validateActionData, zodAddressNonZero } from '@infinit-xyz/core/internal'
+
+import { SetModeStatusSubAction } from '@actions/subactions/setModeStatus'
+import { SetModeStatusTxBuilderParams } from '@actions/subactions/tx-builders/Config/setModeStatus'
+
+import { InitCapitalRegistry } from '@/src/type'
+
+export const SetModeStatusActionParamsSchema = z.object({
+  config: zodAddressNonZero.describe(`Address of protocol config e.g. '0x123...abc'`),
+  modeStatuses: z.custom<Omit<SetModeStatusTxBuilderParams, 'config'>[]>().describe(`pool config parameters`),
```
Comment (mewpawee) [inline packages/modules/init-capital/src/actions/setModeStatus.ts]: change to `PoolConfig` or `Pool config` parameters

File: packages/modules/init-capital/src/actions/setModeStatus.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  constructor(data: SetModeStatusActionData) {
+    validateActionData(data, SetModeStatusActionParamsSchema, ['guardian'])
+    super(SetModeStatusAction.name, data)
+  }
+
+  protected getSubActions(): SubAction[] {
+    const guardian = this.data.signer['guardian']
+    const setModeStatusActionParams: SetModeStatusActionParams = {
+      config: this.data.params.config,
+      modeStatuses: this.data.params.modeStatuses,
```
Comment (mewpawee) [inline packages/modules/init-capital/src/actions/setModeStatus.ts]: new consistency for modeStatus or modeStatuses

File: packages/modules/init-capital/src/actions/subactions/setModeStatus.test.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  test('test correct name', async () => {
+    expect(SetModeStatusSubAction.name).toStrictEqual('SetModeStatusSubAction')
+  })
+
+  test('test correct calldata', async () => {
+    const params: SetModeStatusSubActionParams = {
+      config: '0xCD399994982B3a3836B8FE81f7127cC5148e9BaE',
+      modeStatuses: [
+        { mode: 1, status: { canCollateralize: true, canDecollateralize: false, canBorrow: true, canRepay: false } },
+        { mode: 2, status: { canCollateralize: false, canDecollateralize: false, canBorrow: false, canRepay: false } },
```
Comment (mewpawee) [inline packages/modules/init-capital/src/actions/subactions/setModeStatus.test.ts]: mode status consistency

**Review by melvinmcrn — APPROVED** [review]

**Pull request merged**